src/pages/experiments/[experimentId]/plots-and-tables/dot-plot/index.jsx

```diff
@@ -71,75 +89,249 @@ const DotPlotPage = (props) => {
     error: cellSetsError,
-    hierarchy,
+    hierarchy: cellSetHierarcy,
+    properties: cellSetProperties,
   } = cellSets;
 
-  useEffect(() => {
-    dispatch(loadPlotConfig(experimentId, plotUuid, plotType));
+  const [moreThanTwoGroups, setMoreThanTwoGroups] = useState(false);
 
-    if (hierarchy.length === 0) dispatch(loadCellSets(experimentId));
+  useEffect(() => {
+    if (!config) dispatch(loadPlotConfig(experimentId, plotUuid, plotType));
+    if (cellSetHierarcy.length === 0) dispatch(loadCellSets(experimentId));
   }, []);
 
+  const previousComparedConfig = useRef(null);
+  const getComparedConfig = (updatedConfig) => _.pick(
+    updatedConfig,
+    ['useMarkerGenes',
+      'nMarkerGenes',
+      'selectedGenes',
+      'selectedCellSet',
+      'selectedPoints'],
+  );
+
+  const hasGroupsToCompare = (baseCluster, filterCluster) => {
+    if (cellSetsLoading || !baseCluster || !filterCluster) return false;
+
+    // filterBy has the shape louvain/louvain-1
+    const [filterRootNode, filterKey] = filterCluster.split('/');
+
+    // If 'All" is chosen for the dropdown, there will always be representation from more than 1 group
+    if (filterRootNode === 'All') return true;
+
+    const filterClusterCellIds = Array.from(cellSetProperties[filterKey].cellIds);
+
+    const baseClusterParent = cellSetHierarcy.find((cellSet) => cellSet.key === baseCluster);
+    const baseClusterKeys = baseClusterParent.children.map((child) => child.key);
+    const baseClusterCellIds = baseClusterKeys.map((key) => cellSetProperties[key].cellIds);
+
+    // Check if there is at least 2 baseClusters with elements common to filterCluster
+    let numIntersections = 0;
+    return baseClusterCellIds.some((baseCellIds) => {
+      if (filterClusterCellIds.some((filterCellId) => baseCellIds.has(filterCellId))) {
+        numIntersections += 1;
+      }
+
+      return numIntersections === 2;
+    });
+  };
+
+  const hasGroups = useMemo(
+    () => hasGroupsToCompare(config?.selectedCellSet, config?.selectedPoints),
+    [config?.selectedCellSet, config?.selectedPoints, cellSetsLoading],
+  );
+
   useEffect(() => {
-    if (config && config.selectedGenes.length === 0 && !genesFetching && !highestDispersionGenes) {
-      const PROPERTIES = ['dispersions'];
-      const tableState = {
-        pagination: {
-          current: 1, pageSize: config.nMarkerGenes, showSizeChanger: true, total: 0,
-        },
-        geneNamesFilter: null,
-        sorter: { field: PROPERTIES[0], columnKey: PROPERTIES[0], order: 'descend' },
-      };
+    if (cellSetsLoading) return;
 
-      dispatch(loadPaginatedGeneProperties(experimentId, PROPERTIES, plotUuid, tableState));
+    // Marker genes calculation needs that the cellIds in groupBy (refer to fn definition)
+    // be represented by more than one groups in filterBy to enable comparison
+    // if (config?.useMarkerGenes && !hasGroupsToCompare(config.selectedCellSet, config.selectedPoints)) {
```

Comment:
remove this

Reply:
The comment? I think it's good information so that people knows why we need to check for this,

Reply:
Oh the commented line of code. Good catch! removed.